### PR TITLE
fix: advanced search indexation

### PR DESCRIPTION
**Pull request by pnal**
**Associated Jira issue:** [PISA25-537](https://oat-sa.atlassian.net/browse/PISA25-537)

This PR aims to simplify usage `IndexDocumentBuilder` `createDocumentFrom*` methods and relates with a fix for multiple Service locator issues happening when advanced search is enabled.

What's done:
- Refactored code to use `IndexDocumentBuilder` via DI container.
- Got rid of using  `IndexService` methods proxying `IndexDocumentBuilder`

Related PRs:
- https://github.com/oat-sa/extension-tao-advanced-search/pull/108
- https://github.com/oat-sa/extension-tao-outcomeui/pull/478

[PISA25-537]: https://oat-sa.atlassian.net/browse/PISA25-537?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: models/classes/search/index/IndexService.php [hunk 1]
```diff
@@ -82,9 +79,10 @@ public function getDocumentBuilder(): IndexDocumentBuilderInterface
      */
     public function createDocumentFromResource(\core_kernel_classes_Resource $resource): IndexDocument
     {
+        trigger_error('Method ' . __METHOD__ . ' is deprecated', E_USER_DEPRECATED);
```
Comment (pnal) [inline models/classes/search/index/IndexService.php:82]: We already have deprecation annotations but added additional runtime deprecation warnings, to be safer deciding remove it in future.

**Review by andreluizmachado — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

**Review by bartlomiejmarszal — COMMENTED** [review]
**Review Checklist**
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful
- [ ] Pull request's target is not `master`
- [ ] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Commits messages are meaningful
- [ ] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

File: models/classes/search/tasks/AddSearchIndexFromArray.php [hunk 2]
```diff
@@ -56,13 +56,11 @@ public function __invoke($params)
         }
         $id = array_shift($params);
         $body = array_shift($params);
-        /** @var IndexService $indexService */
-        $indexService = $this->getServiceLocator()->get(IndexService::SERVICE_ID);
```
Comment (gabrielfs7) [inline models/classes/search/tasks/AddSearchIndexFromArray.php:60]: - [ ] We are probably missing also here https://github.com/oat-sa/tao-core/blob/cf88f07dc5f0a434c693b7195fb0e524f72d6c14/models/classes/search/SearchService.php#L65

**Review by bartlomiejmarszal — APPROVED** [review]

File: models/classes/search/tasks/AddSearchIndexFromArray.php (hunk 2 above)
Comment (pnal) [reply models/classes/search/tasks/AddSearchIndexFromArray.php:60]: it uses another method `IndexService::runIndexing()`  which is finally used somewhere in some script in extension-tao-devtools 
https://github.com/search?q=org%3Aoat-sa+runIndexing%28%29%3B&type=code

**Pull request merged**